batch_job/src/batch_job_mesos.c

```diff
@@ -0,0 +1,379 @@
+#include "batch_job.h"
+#include "batch_job_internal.h"
+#include "debug.h"
+#include "process.h"
+#include "macros.h"
+#include "stringtools.h"
+#include "path.h"
+#include "xxmalloc.h"
+#include "jx.h"
+#include "jx_print.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <signal.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <time.h>
+
+#define FILE_TASK_INFO "task_info"
```

Comment:
How about `.mesos_task_info` and `.mesos_task_state` so that we don't pollute the user's directory view?